packages/customer/src/services/customer-module.ts

```diff
@@ -64,18 +72,191 @@ export default class CustomerModuleService implements ICustomerModuleService {
     )
   }
 
-  @InjectManager("baseRepository_")
   async create(
     data: CustomerTypes.CreateCustomerDTO,
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerDTO>
+
+  async create(
+    data: CustomerTypes.CreateCustomerDTO[],
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerDTO[]>
+
+  @InjectManager("baseRepository_")
+  async create(
+    dataOrArray:
+      | CustomerTypes.CreateCustomerDTO
+      | CustomerTypes.CreateCustomerDTO[],
     @MedusaContext() sharedContext: Context = {}
-  ): Promise<CustomerTypes.CustomerDTO> {
-    const [customer] = await this.customerService_.create([data], sharedContext)
+  ) {
+    const data = Array.isArray(dataOrArray) ? dataOrArray : [dataOrArray]
+    const customer = await this.customerService_.create(data, sharedContext)
+
+    if (Array.isArray(dataOrArray)) {
+      return await this.baseRepository_.serialize<CustomerTypes.CustomerDTO[]>(
+        customer,
+        {
+          populate: true,
+        }
+      )
+    }
 
     return await this.baseRepository_.serialize<CustomerTypes.CustomerDTO>(
-      customer,
+      customer[0],
+      {
+        populate: true,
+      }
+    )
```

Comment:
suggestion: you can create a protected create method and depending on the type of the data it returns the customers or customers[0] and here you always serialize. Just to have the same approach everywhere as we do in other modules

Reply:
Not sure I follow the idea here - you would still have to differentiate the `serialize` method since return type is either CustomerDTO or CustomerDTO[]

Reply:
you can get the serialize of an array always and then apply the ternary to this output

Reply:
fixed in upstream PR